### PR TITLE
fix: 9475 adapter node log request error

### LINKED ISSUE #9475: Adapter Node BODY_SIZE_LIMIT
### Describe the problem

Docs:
- https://kit.svelte.dev/docs/adapter-node#environment-variables-body-size-limit
- https://kit.svelte.dev/docs/hooks#shared-hooks-handleerror

When body size limit is exceeded (e.g. a huge file upload) the server request fails, but `handleError()` in `hooks.server.js` is never called.
The issue with that, is that that external logging services (e.g. Sentry) will never report on this issue.

### Describe the proposed solution

I see two possible solutions.

1. call `handleError` directly (simpler for users, but no idea how that would work)
```js
// packages/adapter-node/src/handler.js:82-92
try {
  request = await getRequest({
  base: origin || get_origin(req.headers),
  request: req,
  bodySizeLimit: body_size_limit
});
} catch (err) {
  // call it here if that is possible
  hooksServerHandleError(err);
  res.statusCode = err.status || 400;
  res.end('Invalid request body');
  return;
}
```

2. expose a seperate handleError function through the adapter config
```js
import adapter from '@sveltejs/adapter-node';
 
export default {
  kit: {
    adapter: adapter({
      // default options are shown
      out: 'build',
      precompress: false,
      envPrefix: '',
      polyfill: true,
      handleError: (err) => {}
    })
  }
};
```

### Alternatives considered

To ignore it and maybe do additional frontend validation, when the request fails.

The problem I see with that, is that with a reverse proxy, error messages might not be passed on in a way, that reflect the actual error that happened on the server. 

I am also a fan of not logging backend errors in the frontend 😉

### Importance

nice to have

### Additional Information

For testing: https://cm.events.crm.tuwien.ac.at/register

- go to step 3 in the register form
- upload an image greater than 15MB
image
image

### PR DESCRIPTION
**Pull request by dreitzner**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

Fixes #9475 
This change will additionally log an error if the getRequest function in the SSR handler fails.
I don't think we should test for an error log 😉
Will log an error similar to:
image

### CODE REVIEW

**Comment by tcc-sejohnson** [comment]
I haven't had time to dig into this, just wanted to point out that you _should_ add a test to assert that the warning is being logged. Here's how: 

https://github.com/sveltejs/kit/blob/b48d58e77a8f807075f875f1f1aea420ccec3760/packages/kit/test/apps/basics/test/test.js#L830-L844

**Review by micabal — APPROVED** [review]

File: .changeset/calm-cougars-learn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/adapter-node': minor
```
Comment (benmccann) [inline .changeset/calm-cougars-learn.md:2]: this change could be considered a feature or a fix. I'm not sure it's big enough for a new minor release, so a patch release is probably more appropriate
```suggestion
'@sveltejs/adapter-node': patch
```

File: .changeset/calm-cougars-learn.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Adapter Node: log getRequest error
```
Comment (benmccann) [inline .changeset/calm-cougars-learn.md:5]: we prefix messages with `fix:` or `feat:`. the whole changelog is for adapter-node, so noting that again wouldn't add anything
```suggestion
fix: log error details when invalid request body is encountered
```

**Comment by benmccann** [comment]
My question with this is whether it would get annoying and fill the logs. I'm not that familiar with how logging is configured in JavaScript. This category of issue doesn't indicate any issue with the application and so I'm not sure that `error` is the appropriate console level to use here.

**Comment by symbolicsorcerer** [comment]
I've had a lot of trouble debugging an app running with the node adapter. There's literally nothing logged, ever. I think we need to go even further than this PR, but this is at least a good start to handle one kind of error condition. I've been googling for quite a while to better understand how to troubleshoot adapter-node apps and have come up empty-handed, and also tried asking on the Discord.

When you have a production application deployed with adapter-node, at some point you're going to want some logging for later troubleshooting. If you want to keep the noise down, we could use a log level or env var. I'm not quite sure what the typical JS idioms are here either, but I don't think complete silence is reasonable.

**Pull request closed without merging**